app/Repositories/Movies.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace App\Repositories;
+
+
+use App\Repositories\Contracts\MoviesInterface;
+use GuzzleHttp\Client;
+
+class Movies implements MoviesInterface
+{
+
+    protected $base_uri = 'https://api.themoviedb.org/3/';
+    protected $client;
+
+    public function __construct()
+    {
+        $this->client = new Client();
+    }
+
+
+    protected function fetchAPI($uri, $method = 'GET')
+    {
+        try {
+            $res = $this->client->request($method, $uri);
+            return \GuzzleHttp\json_decode($res->getBody());
+        } catch (\Exception $exception) {
+            return abort($exception->getCode());
+        }
+    }
+
+    public function getUpcomingMovies(int $page = 1)
+    {
+
+        $uri = $this->base_uri . 'movie/upcoming?api_key=' . env('TMDB_APIKEY') . '&language=en-US&page=' . $page;
```

Comment:
string interpolation here will not work because I'm using the return of env function. If exists a way of performing this will be glad to know.
In the past, and even these days (myself included until now), will argue in favor of using . and singles quotes because are faster, but with a small research we can discard this.
- http://phpbench.com/